src/plugins/data_view_editor/public/components/data_view_editor_flyout_content.tsx

```diff
@@ -168,133 +176,78 @@ const IndexPatternEditorFlyoutContentComponent = ({
-    useState<RollupIndicesCapsResponse>({});
-  const [matchedIndices, setMatchedIndices] = useState<MatchedIndicesSet>({
-    allIndices: [],
-    exactMatchedIndices: [],
-    partialMatchedIndices: [],
-    visibleIndices: [],
-  });
 
-  // load all data sources and set initial matchedIndices
-  const loadSources = useCallback(() => {
-    dataViews
-      .getIndices({
-        isRollupIndex: () => false,
-        pattern: '*',
-        showAllIndices: allowHidden,
-      })
-      .then((dataSources) => {
-        setAllSources(dataSources);
-        const matchedSet = getMatchedIndices(dataSources, [], [], allowHidden);
-        setMatchedIndices(matchedSet);
-        setIsLoadingSources(false);
-      });
-  }, [allowHidden, dataViews]);
+  const isLoadingSources = useObservable(dataViewEditorService.isLoadingSources$, true);
 
-  // loading list of index patterns
-  useEffect(() => {
-    loadSources();
-    const getTitles = async () => {
-      const dataViewListItems = await dataViews.getIdsWithTitle(editData ? true : false);
-      const indexPatternNames = dataViewListItems.map((item) => item.name || item.title);
+  const loadingMatchedIndices$ = useRef(new BehaviorSubject<boolean>(false));
 
-      setExistingIndexPatterns(
-        editData ? indexPatternNames.filter((v) => v !== editData.name) : indexPatternNames
-      );
-      setIsLoadingIndexPatterns(false);
-    };
-    getTitles();
-  }, [http, dataViews, editData, loadSources]);
+  const isLoadingDataViewNames$ = useRef(new BehaviorSubject<boolean>(true));
+  const existingDataViewNames$ = useRef(new BehaviorSubject<string[]>([]));
+  const isLoadingDataViewNames = useObservable(isLoadingDataViewNames$.current, true);
 
-  // loading rollup info
+  const rollupIndicesCapabilities = useObservable(
+    dataViewEditorService.rollupIndicesCapabilities$,
+    {}
+  );
+
+  const rollupIndex$ = useRef(new BehaviorSubject<string | undefined>(undefined));
+
+  // initial loading of indicies and data view names
   useEffect(() => {
-    const getRollups = async () => {
-      try {
-        const response = await http.get<RollupIndicesCapsResponse>('/api/rollup/indices');
-        if (response) {
-          setRollupIndicesCapabilities(response);
-        }
-      } catch (e) {
-        // Silently swallow failure responses such as expired trials
-      }
-    };
+    const matchedIndiceSub = dataViewEditorService.matchedIndices$.subscribe((matchedIndices) => {
+      //
+      const timeFieldQuery = editData ? editData.title : title;
+      dataViewEditorService.loadTimestampFields(
+        removeSpaces(timeFieldQuery),
+        type,
+        requireTimestampField,
+        rollupIndex$.current.getValue()
+      );
+    });
 
-    getRollups();
-  }, [http, type]);
+    dataViewEditorService.loadIndices(title, allowHidden).then((matchedIndices) => {
+      dataViewEditorService.matchedIndices$.next(matchedIndices);
```

Comment:
I think we might end up with stale data here. better to check, for example, using `isCanceled` flag if this result is still relevant 
(set `isCanceled = true` in the effect clean up function) 

same for `dataViewEditorService.loadDataViewNames(title)`